[Bug]: Core error on settling a market

### Problem encountered

Whilst setting up market sim to handle successor markets in fuzzing tests, I have hit a couple of core crashing errors at point of settling a market. These could be related to each other, and don't immediately appear to have any relationship to the actual successor markets changes.

### Observed behaviour

When settling a market, the core crashes and exits.

On commit d79ae49de1cdd3d7ebc40efb38d5c6c4f82eefb3 this error was encountered:

```
panic: settlement balance is not zero

goroutine 31149 [running]:
go.uber.org/zap/zapcore.CheckWriteAction.OnWrite(0x80?, 0x1?, {0x1?, 0x0?, 0x0?})
        /Users/tom/go/pkg/mod/go.uber.org/zap@v1.24.0/zapcore/entry.go:198 +0x7c
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0x140040349c0, {0x1400caf2580, 0x1, 0x1})
        /Users/tom/go/pkg/mod/go.uber.org/zap@v1.24.0/zapcore/entry.go:264 +0x2bc
go.uber.org/zap.(*Logger).Panic(0x2?, {0x1030f81a6?, 0x0?}, {0x1400caf2580, 0x1, 0x1})
        /Users/tom/go/pkg/mod/go.uber.org/zap@v1.24.0/logger.go:258 +0x54
code.vegaprotocol.io/vega/core/collateral.(*Engine).CheckLeftOverBalance(0x140014cd050, {0x1041cff80, 0x14009b04690}, 0x1400b720be0, {0x14002328900, 0x2a, 0x40?}, {0x14003cd4cc0, 0x40})
        /Users/tom/Code/vega-market-sim/extern/vega/core/collateral/engine.go:646 +0x84c
code.vegaprotocol.io/vega/core/collateral.(*Engine).FinalSettlement(0x140014cd050, {0x1041cff80, 0x14009b04690}, {0x14003cd5040, 0x40}, {0x14002328900, 0x2a, 0x42})
        /Users/tom/Code/vega-market-sim/extern/vega/core/collateral/engine.go:829 +0x19cc
code.vegaprotocol.io/vega/core/execution/future.(*Market).closeMarket(0x14003aa2a00, {0x1041cff80, 0x14009b04690}, {0x1400012a078?, 0x14005ff4848?, 0x0?})
        /Users/tom/Code/vega-market-sim/extern/vega/core/execution/future/market.go:837 +0x1a4
code.vegaprotocol.io/vega/core/execution/future.(*Market).settlementDataWithLock(0x14003aa2a00, {0x1041cff80, 0x14009b04690})
        /Users/tom/Code/vega-market-sim/extern/vega/core/execution/future/market.go:3349 +0x84
code.vegaprotocol.io/vega/core/execution/future.(*Market).settlementData(0x10?, {0x1041cff80?, 0x14009b04690?}, 0x2?)
        /Users/tom/Code/vega-market-sim/extern/vega/core/execution/future/market.go:3339 +0x108
code.vegaprotocol.io/vega/core/products.(*Future).updateSettlementData(0x14004118000, {0x1041cff80, 0x14009b04690}, {{0x1400e5c87d8?, 0x14006c0f8f0?, 0x14006c0f860?}, 0x14009b046c0?})
        /Users/tom/Code/vega-market-sim/extern/vega/core/products/future.go:254 +0x448
code.vegaprotocol.io/vega/core/oracles.(*Engine).BroadcastData(0x140015fc960, {0x1041cff80, 0x14009b04690}, {{0x1400e5c87d8?, 0x20?, 0x40?}, 0x14009b046c0?})
        /Users/tom/Code/vega-market-sim/extern/vega/core/oracles/engine.go:118 +0x63c
code.vegaprotocol.io/vega/core/processor.(*App).DeliverSubmitOracleData(0x1400147c000, {0x1041cff80, 0x14009b04690}, {0x1041f0870, 0x1400b240800})
        /Users/tom/Code/vega-market-sim/extern/vega/core/processor/abci.go:1704 +0x1cc
```


On version 3685d72393c74fc6b1ad1011a6be9cb7aaa3ff0a this error was encountered:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x101cfe4b0]

goroutine 58102 [running]:
code.vegaprotocol.io/vega/core/collateral.(*Engine).getLedgerEntries(0x14001536630, {0x104773470, 0x14008294360}, 0x14004327410)
        /Users/tom/Code/vega-market-sim/extern/vega/core/collateral/engine.go:2411 +0xc40
code.vegaprotocol.io/vega/core/collateral.(*Engine).clearRemainingLPFees(0x14001536630, {0x104773470, 0x14008294360}, {0x1400d238bc0, 0x40}, {0x1400d238800, 0x40}, 0x1)
        /Users/tom/Code/vega-market-sim/extern/vega/core/collateral/engine.go:2655 +0x5dc
code.vegaprotocol.io/vega/core/collateral.(*Engine).ClearMarket(0x14001536630, {0x104773470, 0x14008294360}, {0x1400d238bc0, 0x40}, {0x1400d238800, 0x40}, {0x1400f93d000, 0x1f, 0x1400070e000?}, ...)
        /Users/tom/Code/vega-market-sim/extern/vega/core/collateral/engine.go:2594 +0x2020
code.vegaprotocol.io/vega/core/execution/future.(*Market).cleanMarketWithState(0x14006fb6a00, {0x104773470, 0x14008294360}, 0x9)
        /Users/tom/Code/vega-market-sim/extern/vega/core/execution/future/market.go:829 +0x1f4
code.vegaprotocol.io/vega/core/execution/future.(*Market).closeMarket(0x14006fb6a00, {0x104773470, 0x14008294360}, {0x140001a8078?, 0x14004328848?, 0x0?})
        /Users/tom/Code/vega-market-sim/extern/vega/core/execution/future/market.go:899 +0x3fc
code.vegaprotocol.io/vega/core/execution/future.(*Market).settlementDataWithLock(0x14006fb6a00, {0x104773470, 0x14008294360})
        /Users/tom/Code/vega-market-sim/extern/vega/core/execution/future/market.go:3735 +0x84
code.vegaprotocol.io/vega/core/execution/future.(*Market).settlementData(0x10?, {0x104773470?, 0x14008294360?}, 0x2?)
        /Users/tom/Code/vega-market-sim/extern/vega/core/execution/future/market.go:3725 +0x108
code.vegaprotocol.io/vega/core/products.(*Future).updateSettlementData(0x1400bfcd290, {0x104773470, 0x14008294360}, {{0x1400ef13a10?, 0x14008120270?, 0x14008120210?}, 0x14008294390?})
        /Users/tom/Code/vega-market-sim/extern/vega/core/products/future.go:254 +0x448
code.vegaprotocol.io/vega/core/oracles.(*Engine).BroadcastData(0x140015ae810, {0x104773470, 0x14008294360}, {{0x1400ef13a10?, 0x20?, 0x40?}, 0x14008294390?})
        /Users/tom/Code/vega-market-sim/extern/vega/core/oracles/engine.go:118 +0x63c
code.vegaprotocol.io/vega/core/processor.(*App).DeliverSubmitOracleData(0x14000160300, {0x104773470, 0x14008294360}, {0x1047941f0, 0x140080bfbc0})
        /Users/tom/Code/vega-market-sim/extern/vega/core/processor/abci.go:1704 +0x1cc
```

In the attached zip, replay corresponds to the first error and replay-2 to the second
[replays.zip](https://github.com/vegaprotocol/vega/files/11960311/replays.zip)



### Expected behaviour

The core not crashing

### Steps to reproduce

Run replay files with the following command:

```bash
vega start --nodewallet-passphrase-file=/path/to/file --home=/path/to/home --blockchain.nullchain.replay --blockchain.nullchain.replay-file=/path/to/replay
```


### Software version

3685d72393c74fc6b1ad1011a6be9cb7aaa3ff0a

### Failing test

_No response_

### Jenkins run

_No response_

### Configuration used

_No response_

### Relevant log output

_No response_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -151,6 +151,7 @@
 - [8665](https://github.com/vegaprotocol/vega/issues/8665) - `Datanode` not persisting stop order events
 - [8702](https://github.com/vegaprotocol/vega/issues/8702) - Fix panic on auction exit after stop orders expired in auction
 - [8703](https://github.com/vegaprotocol/vega/issues/8703) - Wire stop orders cancellation
+- [8698](https://github.com/vegaprotocol/vega/issues/8698) - Always set the `ToAccount` field when clearing fees.
 
 ## 0.71.0
 

--- a/core/collateral/engine.go
+++ b/core/collateral/engine.go
@@ -2616,7 +2616,7 @@ func (e *Engine) clearRemainingLPFees(ctx context.Context, mktID, asset string, 
 	ret := make([]*types.LedgerMovement, 0, 4)
 	req := &types.TransferRequest{
 		FromAccount: make([]*types.Account, 1),
-		ToAccount:   make([]*types.Account, 1),
+		ToAccount:   []*types.Account{marketInsuranceAcc},
 		Asset:       asset,
 		Type:        types.TransferTypeClearAccount,
 	}
@@ -2624,7 +2624,6 @@ func (e *Engine) clearRemainingLPFees(ctx context.Context, mktID, asset string, 
 	lpFeeAcc, exists := e.accs[lpFeeAccID]
 	if exists && !lpFeeAcc.Balance.IsZero() {
 		req.FromAccount[0] = lpFeeAcc
-		req.ToAccount[0] = marketInsuranceAcc
 		req.Amount = lpFeeAcc.Balance.Clone()
 		lpFeeLE, err := e.getLedgerEntries(ctx, req)
 		if err != nil {

--- a/core/collateral/engine_test.go
+++ b/core/collateral/engine_test.go
@@ -125,6 +125,64 @@ func TestTransferRewards(t *testing.T) {
 	t.Run("transfer rewards success", testTransferRewardsSuccess)
 }
 
+func TestClearAccounts(t *testing.T) {
+	t.Run("clear fee accounts", testClearFeeAccounts)
+}
+
+func testClearFeeAccounts(t *testing.T) {
+	eng := getTestEngine(t)
+	defer eng.Finish()
+	ctx := context.Background()
+	eng.broker.EXPECT().Send(gomock.Any()).AnyTimes()
+	mktID := "market"
+	asset := "ETH"
+	party := "myparty"
+	assetT := types.Asset{
+		ID: asset,
+		Details: &types.AssetDetails{
+			Symbol: asset,
+		},
+	}
+
+	eng.EnableAsset(ctx, assetT)
+	_, _ = eng.GetGlobalRewardAccount(asset)
+	_, _, err := eng.CreateMarketAccounts(ctx, mktID, asset)
+	require.NoError(t, err)
+	general, err := eng.CreatePartyGeneralAccount(ctx, party, asset)
+	require.NoError(t, err)
+
+	_, err = eng.CreatePartyMarginAccount(ctx, party, mktID, asset)
+	require.NoError(t, err)
+
+	// add funds
+	err = eng.UpdateBalance(ctx, general, num.NewUint(10000))
+	assert.Nil(t, err)
+
+	transferFeesReq := transferFees{
+		tfs: []*types.Transfer{
+			{
+				Owner: party,
+				Amount: &types.FinancialAmount{
+					Amount: num.NewUint(1000),
+				},
+				Type:      types.TransferTypeMakerFeePay,
+				MinAmount: num.NewUint(1000),
+			},
+		},
+		tfa: map[string]uint64{party: 1000},
+	}
+
+	transfers, err := eng.TransferFeesContinuousTrading(ctx, mktID, asset, transferFeesReq)
+	assert.NotNil(t, transfers)
+	assert.NoError(t, err, collateral.ErrInsufficientFundsToPayFees.Error())
+	assert.Len(t, transfers, 1)
+
+	assert.Equal(t, 1, len(transfers[0].Entries))
+	assert.Equal(t, num.NewUint(9000), transfers[0].Entries[0].FromAccountBalance)
+	assert.Equal(t, num.NewUint(1000), transfers[0].Entries[0].ToAccountBalance)
+	eng.ClearInsurancepool(ctx, mktID, asset, true)
+}
+
 func testTransferRewardsEmptySlice(t *testing.T) {
 	eng := getTestEngine(t)
 	defer eng.Finish()

